Mathlib/Data/Polynomial/Derivative.lean

```diff
@@ -669,6 +680,10 @@ theorem iterate_derivative_X_sub_pow (n k : ℕ) (c : R) :
 set_option linter.uppercaseLean3 false in
 #align polynomial.iterate_derivative_X_sub_pow Polynomial.iterate_derivative_X_sub_pow
 
+theorem iterate_derivative_X_sub_pow' (n k : ℕ) (c : R) :
+    (derivative^[k]) ((X - C c) ^ n : R[X]) = n.descFactorial k • (X - C c) ^ (n - k) := by
+  rw [iterate_derivative_X_sub_pow, Nat.descFactorial_eq_prod_range, nsmul_eq_mul]
```

Comment:
`iterate_derivative_X_add_pow`, `iterate_derivative_X_sub_pow`, `coeff_iterate_derivative_as_prod_range` are not used anywhere, so I guess we could just change their statements?

I haven't looked at Derivative2.lean carefully, but some stuff there look like they could be PR'd into this file ...